### PR TITLE
Policy Manager Impl Tests Fix

### LINKED ISSUE #517: Policy Manager Impl unit tests are omitted from build
[Line omitted here](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/policy/test/CMakeLists.txt#L56)

After uncommenting that line and including that file in the build to be tested, I was not surprised to see that the test did not pass.

The policy table created is not valid.

Here are some of the tests logs highlighting why the test failed

```
[ RUN      ] PolicyManagerImplTest.LoadPT_SetPT_PTIsLoaded
/home/jack/Projects/sdl_core/src/components/policy/test/policy_manager_impl_test.cc:260: Failure
Value of: IsValid(update)
  Actual: false ( - table.policy_table.app_policies.device: object is not initialized
 - table.policy_table.app_policies.device.groups: object is not initialized
 - table.policy_table.app_policies.device.priority: value is not initialized
 - table.policy_table.app_policies.device.default_hmi: value is not initialized
 - table.policy_table.app_policies.device.keep_context: value is not initialized
 - table.policy_table.app_policies.device.steal_focus: value is not initialized
)
Expected: true
[  FAILED  ] PolicyManagerImplTest.LoadPT_SetPT_PTIsLoaded (2 ms)
[ RUN      ] PolicyManagerImplTest.RequestPTUpdate_SetPT_GeneratedSnapshotAndPTUpdate
ERROR [27 Apr 2016 11:54:26,844][PolicyManagerImpl] Policy table is not valid.
```

### PR DESCRIPTION
**Pull request by JackLivio**
Uncommented out policy tests so policy unit tests are included in the build. Also fixed one of the test cases to include the necessary parts for app_policies in the 'LoadPT_SetPT_PTIsLoaded' test. "make test" now passes.

Fixes #517

### CODE REVIEW

File: src/components/policy/test/policy_manager_impl_test.cc [hunk 1]
```diff
@@ -253,6 +254,22 @@ TEST_F(PolicyManagerImplTest, LoadPT_SetPT_PTIsLoaded) {
   app_policies["1234"]["steal_focus"] = Json::Value(true);
   app_policies["1234"]["certificate"] = Json::Value("sign");
 
+  app_policies["device"] = Json::Value(Json::objectValue);
+  app_policies["device"]["keep_context"] = Json::Value(false);
```
Comment (AGaliuzov) [inline src/components/policy/test/policy_manager_impl_test.cc]: @JackLivio these values should be omited here. So the `steal_focus`, `keep_context`, `defalt_hmi` are shoul not be presented here at all. And this is what  #526 print out

Comment (JackLivio) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AGaliuzov Why should these values be omitted? Removing "device" from app policies or its keep_context, default_hmi, or steal_focus values causes the policy table to be considered invalid.


Results if values are removed:
```
policy_manager_impl_test.cc:277: Failure
Value of: IsValid(update)
  Actual: false ( - table.policy_table.app_policies.device: object is not initialized
 - table.policy_table.app_policies.device.groups: object is not initialized
 - table.policy_table.app_policies.device.priority: value is not initialized
 - table.policy_table.app_policies.device.default_hmi: value is not initialized
 - table.policy_table.app_policies.device.keep_context: value is not initialized
 - table.policy_table.app_policies.device.steal_focus: value is not initialized
)
```

Comment (JackLivio) [inline src/components/policy/test/policy_manager_impl_test.cc]: @AGaliuzov Also your link to what is printed out is a link back to this issue?

Comment (AGaliuzov) [inline src/components/policy/test/policy_manager_impl_test.cc]: @JackLivio Sorry I have mentioned the ticket #517. These value should be omitted in in Open source due to current Ford's requirements

**Comment by LuxoftAKutsan** [comment]
@JackLivio  do you agree with @AGaliuzov  that this values should be omitted in GENIVI? If yes, probably you should close this pull request

**Comment by JackLivio** [comment]
@AGaliuzov @LuxoftAKutsan  If these values should be omitted in the open source version of sdl core then how come these values are [`included in the preloaded policy table? `](https://github.com/smartdevicelink/sdl_core/blob/master/src/appMain/sdl_preloaded_pt.json#L2248-L2270)

However it appears this test file has changed a lot in the recent 4.1 merge and that this PR is no longer valid. I will close this.

**Pull request closed without merging**